app/javascript/react/components/ReviewForm.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useEffect } from "react";
+import { StarRating } from "@thumbtack/thumbprint-react";
+
+const ReviewForm = (props) => {
+
+  let flavorId = props.match.params.flavor_id
+
+  // These 4 states hold each of our star ratings for review attributes
+  const [overallRating, setOverallRating] = useState(null);
+  const [sweetnessRating, setSweetnessRating] = useState(null);
+  const [mouthFeelRating, setMouthFeelRating] = useState(null);
+  const [tasteRating, setTasteRating] = useState(null);
+
+  // These 4 states hold hover values for each review attribute
+  const [overallHoverRating, setOverallHoverRating] = useState(undefined);
+  const [sweetnessHoverRating, setSweetnessHoverRating] = useState(undefined);
+  const [mouthFeelHoverRating, setMouthFeelHoverRating] = useState(undefined);
+  const [tasteHoverRating, setTasteHoverRating] = useState(undefined);
+
+  // This complex state holds all other review attributes and receives the state from star ratings attributes before submission
+  const [reviewData, setReviewData] = useState({
+      overall: null,
+      sweetness: null,
+      mouth_feel: null,
+      taste: null,
+      comment: "",
+      manufacturer: ""
+  })
+
+  const [getNotice, setNotice] = useState("")
+
+  const handleTextInputChange = (event) => {
+    setReviewData({
+      ...reviewData,
+      [event.currentTarget.name]: event.currentTarget.value,
+      overall: overallRating,
+      sweetness: sweetnessRating,
+      mouth_feel: mouthFeelRating,
+      taste: tasteRating,
+    });
+  };
+
+  const handleSubmit = () => {
+    event.preventDefault();
+    fetch(`/api/v1/flavors/${flavorId}/reviews`, {
+      method: "POST",
+      credentials: "same-origin",
+      body: JSON.stringify(reviewData),
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+    })
+      .then(response => {
+        if (response.ok) {
+          return response
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw error
+        }
+      })
+      .then(response => response.json())
+      .then(body => {
+        setNotice(body.notice)
+        body 
+      })
+      .catch(error => console.error(`Error in fetch: ${error.message}`))
+  };
+
+  return (
+    <div>
+      {getNotice} <br/>
+      <form onSubmit={handleSubmit} id="reviewForm">
+        <h5>Overall</h5>
+        <StarRating
+          size="large"
+          name="overall"
+          rating={overallRating}
+          hoverRating={overallHoverRating}
+          onStarClick={(value) => {
+            setOverallRating(value)
+          }}
```

Comment:
ideally, for any of these onStarClicks, we use a callback function that in turn uses `setReviewData` setter. 

Said callback function much like `handleTextInputChange`, might look something like

```
handleRatingChange = (value, ratingName) = { 
      setReviewData({
      ...reviewData,
      [ratingName]: value
  })
}
```

In this way, we dont need to track as much state, though it does seem like you need the hover state

Reply:
Resolved in above comment with your changes.